imap_processing/mag/l1a/mag_l1a.py

```diff
@@ -25,22 +29,76 @@ def mag_l1a(packet_filepath):
     norm_data = packets["norm"]
     burst_data = packets["burst"]
 
-    if norm_data is not None:
-        mag_norm_raw = decom_mag.generate_dataset(
-            norm_data, mag_cdf_attrs.mag_l1a_norm_raw_attrs.output()
-        )
-        file = write_cdf(mag_norm_raw)
-        logger.info(f"Created RAW CDF file at {file}")
+    generated_files = process_and_write_data(
+        norm_data,
+        mag_cdf_attrs.mag_l1a_norm_raw_attrs.output(),
+        mag_cdf_attrs.mag_l1a_norm_mago_attrs.output(),
+        mag_cdf_attrs.mag_l1a_norm_magi_attrs.output(),
+    )
+    generated_files += process_and_write_data(
+        burst_data,
+        mag_cdf_attrs.mag_l1a_burst_raw_attrs.output(),
+        mag_cdf_attrs.mag_l1a_burst_mago_attrs.output(),
+        mag_cdf_attrs.mag_l1a_burst_magi_attrs.output(),
+    )
 
-    if burst_data is not None:
-        mag_burst_raw = decom_mag.generate_dataset(
-            burst_data, mag_cdf_attrs.mag_l1a_burst_raw_attrs.output()
-        )
-        file = write_cdf(mag_burst_raw)
-        logger.info(f"Created RAW CDF file at {file}")
+    return generated_files
 
 
-def process_packets(mag_l0_list: list[MagL0]) -> dict[str, list[MagL1a]]:
+def process_and_write_data(
+    packet_data: list[MagL0], raw_attrs: dict, mago_attrs: dict, magi_attrs: dict
+):
+    """
+    Process MAG L0 data into L1A, then create and write out CDF files.
+
+    Norm and burst mode descriptors are distinguished with the passed in attrs.
+
+    Parameters
+    ----------
+    packet_data: list[MagL0]
+        List of MagL0 packets to process, containing primary and secondary sensor data
+    raw_attrs
+        Attributes for MagL1A raw CDF files
+    mago_attrs
+        Attributes for MagL1A MAGo CDF files
+    magi_attrs
+        Attributes for MagL1A MAGi CDF files
+
+    Returns
+    -------
+    generated_files: list[str]
+        A list of generated filenames
+    """
+    if not packet_data:
+        return []
+
+    mag_raw = decom_mag.generate_dataset(packet_data, raw_attrs)
```

Comment:
can you call attrs here instead of passed in? Just curious
```suggestion
    mag_raw = decom_mag.generate_dataset(packet_data, mag_cdf_attrs.mag_l1a_burst_raw_attrs.output())
```

Reply:
Norm and burst use different attrs, so unfortunately not.